### PR TITLE
CKAN 2.10 replace controllers & IRoutes with blueprints

### PR DESCRIPTION
**Pull request by thepsalmist**
Migration of all the controllers to flask views/blueprints. This means changing all BaseController controllers handled by the IRoutes interface to blueprints handled by IBlueprint interface.

IRoutes is deprecates in CKAN < 2.10

### CODE REVIEW

File: ckanext/openafrica/blueprint.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    return render('home/about/privacy.html')
+
+def contact():
+    return render('home/about/contact.html')
+
+def atlas():
+    return render('atlas.html')
+
+rules = [
+    ("/about/terms-and-conditions", "toc", toc),
```
Comment (kilemensi) [inline ckanext/openafrica/blueprint.py]: Unless I'm missing something, a closure would work better here:
```python
def static_path(path):
    def render_path():
        return render(path)

    return render_path
```

And we'd use it as:
```python
rules = [
    ("/about/terms-and-conditions", "toc", static_path('home/about/toc.html')),
    ("/about/accessibility", "accessibility", static_path('home/about/accessibility.html')),
    ....
```

1. May be you can think of a better, shorter name that static_path,
2. Don't mix use of single and double quotes; pick one and stick with it.

File: ckanext/openafrica/plugin.py [hunk 2]
```diff
@@ -23,15 +23,16 @@
 
 import ckan.plugins as plugins
 import ckan.plugins.toolkit as toolkit
+from . import blueprint
```
Comment (kilemensi) [inline ckanext/openafrica/plugin.py:26]: Prefer absolute imports.

File: ckanext/openafrica/plugin.py [hunk 3]
```diff
@@ -23,15 +23,16 @@
...
 class OpenAfricaPlugin(plugins.SingletonPlugin):
     u"""
     openAFRICA templating plugin.
     """
-    plugins.implements(plugins.IConfigurer)
-    plugins.implements(plugins.IRoutes, inherit=True)
-    plugins.implements(plugins.ITemplateHelpers)
+    plugins.implements(plugins.IConfigurer, inherit=True)
+    plugins.implements(plugins.ITemplateHelpers, inherit=True)
+    plugins.implements(plugins.IBlueprint)
```
Comment (kilemensi) [inline ckanext/openafrica/plugin.py:35]: Why do some plugins use inherit and others don't? (This is for just my understanding).

**Review by koechkevin — APPROVED** [review]

File: ckanext/openafrica/plugin.py (hunk 3 above)
Comment (thepsalmist) [reply ckanext/openafrica/plugin.py:35]: On the blueprint, we only need custom routes to handle our `/about/<>/` urls hence though it was not necessary to extend CKAN core's Blueprints. On the other plugins, we inherit helper functions & configurations  such as `url_for`, `link_for` et.al from CKAN core

**Pull request merged**